.github/workflows/twister.yaml

```diff
@@ -259,22 +259,41 @@ jobs:
           if-no-files-found: ignore
           path: |
             twister-out/twister.xml
+            twister-out/twister.json
             module_tests/twister.xml
             testplan.json
 
   twister-test-results:
     name: "Publish Unit Tests Results"
+    env:
+      OPENSEARCH_USER: ${{ secrets.OPENSEARCH_USER }}
+      OPENSEARCH_PASS: ${{ secrets.OPENSEARCH_PASS }}
     needs: twister-build
     runs-on: ubuntu-20.04
       # the build-and-test job might be skipped, we don't need to run this job then
     if: success() || failure()
 
     steps:
+      # Needed for opensearch and upload script
+      - if: github.event_name == 'push'
+        name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
should not use `github.event.pull_request.head.sha` for push events. `ref` here can be omitted.